components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/utils/SelfSignUpUtil.java

```diff
@@ -18,184 +18,31 @@
-	 *         tenant
-	 * @throws APIManagementException
-	 */
-	public static UserRegistrationConfigDTO getSignupConfiguration(String tenantDomain)
-			throws APIManagementException {
-
-		return getSignupConfigurationFromRegistry(tenantDomain);
-	}
 
 	/**
-	 * load configuration from the registry
-	 * 
-	 * @param tenantDomain - The Tenant Domain
-	 * @return - A UserRegistrationConfigDTO instance
+	 * Retrieve self signup configuration from the Advanced Configuration and return the configuration
+	 *
+	 * @param tenantDomain Domain name of the tenant
+	 * @return UserRegistrationConfigDTO self signup configuration for the tenant
 	 * @throws APIManagementException
 	 */
-	private static UserRegistrationConfigDTO getSignupConfigurationFromRegistry(String tenantDomain)
-			throws APIManagementException {
-
-		UserRegistrationConfigDTO config;
+	public static UserRegistrationConfigDTO getSignupConfiguration(String tenantDomain) throws APIManagementException {
 
-		try {
-			String selfSighupConfig =
-					ServiceReferenceHolder.getInstance().getApimConfigService().getSelfSighupConfig(tenantDomain);
-			OMElement element = AXIOMUtil.stringToOM(selfSighupConfig);
-			config = new UserRegistrationConfigDTO();
-			config.setSignUpDomain(element.getFirstChildWithName(
-					new QName(APIConstants.SELF_SIGN_UP_REG_DOMAIN_ELEM)).getText());
-			config.setAdminUserName(APIUtil.replaceSystemProperty(
-					element.getFirstChildWithName(new QName(
-							APIConstants.SELF_SIGN_UP_REG_USERNAME)).getText()));
-			String encryptedPassword = element
-					.getFirstChildWithName(new QName(APIConstants.SELF_SIGN_UP_REG_PASSWORD)).getText();
-			PasswordResolver passwordResolver = PasswordResolverFactory.getInstance();
-			String resovledPassword = passwordResolver.getPassword(encryptedPassword);
-			config.setAdminPassword(APIUtil.replaceSystemProperty(resovledPassword));
-			config.setSignUpEnabled(Boolean.parseBoolean(element.getFirstChildWithName(
-					new QName(APIConstants.SELF_SIGN_UP_REG_ENABLED)).getText()));
-			OMElement rolesElement =
-					element.getFirstChildWithName(new QName(APIConstants.SELF_SIGN_UP_REG_ROLES_ELEM));
-			Iterator roleListIterator = rolesElement.getChildrenWithLocalName(APIConstants.SELF_SIGN_UP_REG_ROLE_ELEM);
-			while (roleListIterator.hasNext()) {
-				OMElement roleElement = (OMElement) roleListIterator.next();
-				String tmpRole = roleElement.getFirstChildWithName(
-						new QName(APIConstants.SELF_SIGN_UP_REG_ROLE_NAME_ELEMENT)).getText();
-				boolean tmpIsExternal = Boolean.parseBoolean(roleElement.getFirstChildWithName(
-						new QName(APIConstants.SELF_SIGN_UP_REG_ROLE_IS_EXTERNAL)).getText());
-				config.getRoles().put(tmpRole, tmpIsExternal);
-			}
-		} catch (XMLStreamException e) {
-			throw new APIManagementException("Error while parsing configuration ", e);
+		Object selfSighupConfigObject = ServiceReferenceHolder.getInstance().getApimConfigService()
```

Comment:
Typo in variable name

Reply:
Fixed in https://github.com/wso2/carbon-apimgt/pull/11919